code/game/objects/effects/decals/misc.dm

```diff
@@ -89,3 +89,14 @@
 	icon = 'icons/obj/smooth_structures/lattice.dmi'
 	icon_state = "lattice-255"
 	density = TRUE
+
+/// Given an atom, will return what color it should be to match the event/holiday
+/obj/effect/proc/get_holiday_color(atom/thing_to_color, axis = STRIPE_HORIZONTAL)
```

Comment:
I'm not sure you want this to be a proc on `/obj/effect`, given it's copy+pasted onto `/turf`

Maybe you should instead bump it down a level from `/datum/holiday/pride_week/proc/get_floor_tile_color` to `/datum/holiday/proc/get_floor_tile_color`, which then each holiday can override on a subtype basis to do their own custom coloring if desired

Reply:
The decals are /objects/effects/decals/turfdecal, not a turf. My thought behind the global color list is its also used by the rainbow tile objects (which are rainbow year round, not just the one week.)

For other holidays they can override the list with their holiday specific colors.